tests/units/aps/test_aps_file_processing.py

```diff
@@ -0,0 +1,43 @@
+import json
+
+from aps.aps_process_file import aps_validate_record, enhance_aps, enrich_aps
+from aps.parser import APSParser
+from pytest import fail, fixture
+
+
+@fixture
+def articles(shared_datadir):
+    json_response = (shared_datadir / "json_response_content.json").read_text()
+    return [article for article in json.loads(json_response)["data"]]
+
+
+@fixture
+def parser():
+    return APSParser()
+
+
+@fixture
+def parsed_articles(articles, parser):
+    return [parser._publisher_specific_parsing(article) for article in articles]
+
+
+@fixture
+def parsed_generic_articles(parsed_articles, parser):
+    return [parser._generic_parsing(article) for article in parsed_articles]
+
+
+@fixture
+def enhance_articles(parsed_generic_articles):
+    return [enhance_aps(article) for article in parsed_generic_articles]
+
+
+@fixture
+def enrich_article(enhance_articles):
+    return [enrich_aps(article) for article in enhance_articles]
+
+
+def test_aps_validated_record(enrich_article):
+    try:
+        [aps_validate_record(article) for article in enrich_article]
+    except (Exception):
+        fail()
```

Comment:
what this is testing?

Reply:
is the article is correct, after all steps of workflow: it runs validation on article

Reply:
thinking more, I see your point, we don't need it